marshmallow_enum/__init__.py

```diff
@@ -71,6 +71,11 @@ def __init__(
 
         super(EnumField, self).__init__(*args, **kwargs)
 
+        self.metadata['enum'] = [
+            e.value if self.by_value else e.name
```

Comment:
by_value needs to be deprecated in favor of the `load_by` and `dump_by` attributes so this isn't a reliable path forward. The other issue is that you can load by either name/value and dump by name/value independently. I'd rather not deprecate this behavior as it's actually a nice way of working around not great third party APIs (e.g. load from their bad API into our nicer one, and then dump out our nicer version).

Reply:
As far as I can tell, the openapi spec does not allow differentiating between using a schema for loading and for dumping data.

Would it be a reasonable approach to set the `enum` metadata to either the values or the names if both `load_by` and `dump_by` are equal and use both values and names in other cases?

For the test this would result in 

```python
class Test(Enum):
    one = 1
    two = 2
    three = 3
```

| `load_by` | `dump_by` | `metadata['enum']` |
| --------- | --------- | ------------------ |
| `name` | `name` | `one,two,three` |
| `value` | `value` | `1,2,3` |
| `name` | `value` | `one,two,three,1,2,3` |
| `value` | `name` | `one,two,three,1,2,3` |

Reply:
Seems that'd be confusing to end users. As for differentiating between the sent model and returned model, is that a limit of apispec or the marshmallow extension? Because that's a fairly common use case in apis.

Reply:
I think @Birne94 's proposal adds a lot of value for the 

> 99% of the time they'll be the same

and is also my personal preference for the cases where they're not as well (If automatic tooling can't get the exact set of allowable values, at least it narrows it down to a superset which is no more than 2x the set of allowable values for a person who cares about one side of the api--either the ingested schema or the output schema). 

@justanr if you disagree, would not setting it for `load_by != dump_by` cases be preferable? Then it does no harm if it can't get it exactly right, and is helpful in the cases where it definitely can.

Reply:
> Seems that'd be confusing to end users. As for differentiating between the sent model and returned model, is that a limit of apispec or the marshmallow extension? Because that's a fairly common use case in apis.

This is actually a limitation of the OpenAPI specification itself. I'd suggest either raising an error, or falling back on the previous behavior when load_by/dump_by are unequal.